app/controllers/likes_controller.rb

```diff
@@ -0,0 +1,11 @@
+class LikesController < ApplicationController
+  def create
+    like = Post.find(params[:post_id]).likes.new(user: current_user)
+    if like.save
+      flash[:success] = 'Post liked'
+    else
+      flash.now[:error] = 'Error: Failed to like post'
+    end
+    redirect_back(fallback_location: users_path)
+  end
```

Comment:
[OPTIONAL] Use `post.likes.build` instead of `Post.find(params[:post_id]).likes.new`. This is a more idiomatic way in Rails to associate the new record with its parent.

Reply:
Good note thank you!